SandB-iOS/SandB-iOS/Controllers/ArticlesListTableViewController.m

```diff
@@ -92,152 +78,165 @@ - (void)viewWillAppear:(BOOL)animated {
+                                       stringWithFormat:@"%@ | %@", article.date, article.author];
+    } else {
+        cell.articleDetails.text = [NSString stringWithFormat:@"%@ | S&B", article.date];
     }
     
     // if article has been clicked on, aka read, color it with the category color to mark it as read
-    if ([[[DataModel sharedModel] readArticles] containsObject:[self.allArticlesArray objectAtIndex:indexPath.row]] && ![self.newsCategory isEqualToString:@"Favorites"]) {
-        cell.backgroundColor = [[[[NewsCategories sharedCategories] categoriesByName] objectForKey:[[self.allArticlesArray  objectAtIndex:indexPath.row] category]]  highlightedColor];
-        cell.articleTitle.textColor = [UIColor grayColor];
+    if ([[[DataModel sharedModel] readArticles] containsObject:article] &&
+        ![self.newsCategory isEqualToString:@"Favorites"]) {
+        
         [cell.articleTitle setFont:[UIFont fontWithName:@"ProximaNova-Light" size:18]];
+        
+        cell.articleTitle.textColor = [UIColor grayColor];
         cell.articleDetails.textColor = [UIColor grayColor];
-        cell.categoryIdentifier.backgroundColor = [[[[NewsCategories sharedCategories] categoriesByName] objectForKey:[[self.allArticlesArray objectAtIndex:indexPath.row] category]] readBarColor];
         
-    }
-    else {
-        cell.backgroundColor = [UIColor whiteColor];
-        cell.articleTitle.textColor = [UIColor blackColor];
+        cell.backgroundColor = category.highlightedColor;
+        cell.categoryIdentifier.backgroundColor = category.readBarColor;
+        
+    } else {
         [cell.articleTitle setFont:[UIFont fontWithName:@"ProximaNova-Regular" size:18]];
+        
+        cell.articleTitle.textColor = [UIColor blackColor];
         cell.articleDetails.textColor = [UIColor blackColor];
-        cell.categoryIdentifier.backgroundColor = [[[[NewsCategories sharedCategories] categoriesByName] objectForKey:[[self.allArticlesArray objectAtIndex:indexPath.row] category]] color];
+        
+        cell.backgroundColor = [UIColor whiteColor];
+        cell.categoryIdentifier.backgroundColor = category.color;
     }
     
     [cell.articleDetails setFont:[UIFont fontWithName:@"ProximaNova-Light" size:12]];
     
-    // make sure the selected color stays
-    
     return cell;
 }
 
 
-- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
-{
-    if (indexPath.row <  [self.allArticlesArray   count]) {
+- (UITableViewCell *)tableView:(UITableView *)tableView
+         cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    if (indexPath.row <  [self.allArticlesArray count]) {
         return [self articleCellForIndexPath:indexPath];
     } else {
         return [self loadingCell];
     }
 }
 
-- (UITableViewCell *)loadingCell
-{
-    UITableViewCell *cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:nil];
+- (UITableViewCell *)loadingCell {
```

Comment:
This cell is inserted at the bottom of the table view. It's for loading the next "page" of articles, not at the top.